### PR TITLE
[bugfix] Fix --stored-query parameter in goQuery and accept stdin

### PR DESCRIPTION
**Pull request by els0r**
When running a query with `--stored-query` from disk, the argument for the location of the query args wasn't taken into account, rendering the flag unusable.

This commit fixes the parameter and extends functionality: in line with other tooling, if `-` is provided as the location of the query args, `goQuery` assumes that they are read from `stdin`. Thus, arguments can now be piped into `goQuery`.

Only to be merged after #234 is on main.

### CODE REVIEW

File: cmd/goQuery/cmd/root.go [hunk 1]
```diff
@@ -247,14 +247,23 @@ func entrypoint(cmd *cobra.Command, args []string) (err error) {
 
 	// check if arguments should be loaded from disk. The cmdLineParams are taken as
 	// the base for this to allow modification of single parameters
-	if viper.GetString(conf.StoredQuery) != "" {
-		argumentsJSON, err := os.ReadFile(filepath.Clean(argsLocation))
-		if err != nil {
-			return fmt.Errorf("failed to read query args from %s: %w", argsLocation, err)
+	argsLocation := viper.GetString(conf.StoredQuery)
+	if argsLocation != "" {
+		var argsReader io.Reader
```
Comment (fako1024) [inline cmd/goQuery/cmd/root.go:252]: Code simplification: Initialize `argsReader` with `os.Stdin` and omit the `else` statement below:
```
var argsReader io.Reader = os.Stdin
if argsLocation != "-" {
...
}
```

File: pkg/goDB/protocols/protocols_generator.go [hunk 2]
```diff
@@ -66,15 +75,30 @@ func readProtocols() (protoList []string, err error) {
 			continue
 		}
 
-		protoList[protoID] = fields[2]
+		// discard any id that's greater than 255. It would be invalid during the
```
Comment (fako1024) [inline pkg/goDB/protocols/protocols_generator.go:78]: Why is that necessary? AFAIK the OS Proto ID are guaranteed to be 8-bit already... :thinking:

File: pkg/goDB/protocols/protocols_generator.go [hunk 3]
```diff
@@ -47,7 +53,10 @@ func readProtocols() (protoList []string, err error) {
 		}
 	}()
 
-	protoList = make([]string, 256)
+	// auxiliary to filter out duplicates
+	var seen = make(map[string]struct{})
```
Comment (fako1024) [inline pkg/goDB/protocols/protocols_generator.go:57]: Why do we need duplicate filtering? The OS list should not have any (at least none that we don't want to actually track - and since it's a map it should overwrite anyway _if_ there are any duplicates from the key side)?

**Review by fako1024 — CHANGES_REQUESTED** [review]
I'm confused about the protocol handling (after all, this whole thing should _theoretically_ not really be needed anymore because that list is relatively stable), can you maybe elaborate / cross-check?

File: pkg/goDB/protocols/protocols_generator.go (hunk 2 above)
Comment (fako1024) [reply pkg/goDB/protocols/protocols_generator.go:78]: Urgs, I just saw https://github.com/els0r/goProbe/pull/234 - That sucks. What's proto 258 on Darwin anyway? I always thought this is a mere cosmetic thing. Maybe we should rethink the whole handling here after all, we're currently not planning for Darwin, so maybe we should wing that whole discussion / handling until we do (after all, right now we might introduce some weird cross-system incompatibilities and not even be aware of it)?

File: pkg/goDB/protocols/protocols_generator.go (hunk 3 above)
Comment (fako1024) [reply pkg/goDB/protocols/protocols_generator.go:57]: Nvm, #234 ...

**Review by fako1024 — APPROVED** [review]

File: cmd/goQuery/cmd/root.go (hunk 1 above)
Comment (fako1024) [reply cmd/goQuery/cmd/root.go:252]: @els0r I've approved the PR, maybe you could still have a quick look at this tiny nitpick change?

Comment (els0r) [reply cmd/goQuery/cmd/root.go:252]: > @els0r I've approved the PR, maybe you could still have a quick look at this tiny nitpick change?

Never nesting. I hear ya
